### PR TITLE
Convert os.c to os.d

### PR DESCRIPTION
**Pull request by wilzbach**
This probably will still fail on Windows as I'm using auto-tester as my Windows box.

### CODE REVIEW

**Comment by wilzbach** [comment]
Some Windows magic again :/

```
C:\cygwin\home\braddr\sandbox\at-client\release-build\dmd-2.079.0\windows\bin\..\lib\snn.lib(io)  Offset 00B35H Record Type 0091 
 Error 1: Previous Definition Different : __osfhnd
```

File: src/dmd/backend/os.d [hunk 1]
```diff
@@ -628,42 +656,52 @@ int os_file_exists(const char *name)
...
 
 /**************************************
  * Get file size of open file. Return -1L on error.
  */
 
-#if _WIN32 && !_MSC_VER
-extern "C" void * __cdecl _osfhnd[];
-#endif
+//static if (NEEDS_WIN32_NON_MS)
+version (Win32)
```
Comment (rainers) [inline src/dmd/backend/os.d]: Should be `CRuntime_DigitalMars`

File: src/dmd/backend/os.d [hunk 2]
```diff
@@ -628,42 +656,52 @@ int os_file_exists(const char *name)
...
+{
+extern (C) void*[] _osfhnd;
```
Comment (rainers) [inline src/dmd/backend/os.d]: `extern extern(C) void*[] _osfhnd;`

**Review by jacob-carlborg — APPROVED** [review]

**Comment by thewilsonator** [comment]
#8948

**Comment by thewilsonator** [comment]
#8949

**Pull request closed without merging**